### PR TITLE
feat(tokens): migrate user added data from legacy state

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

During the refactoring, some data was moved from redux to jotai atoms.

1. User-added tokens: `redux_localstorage_simple_user.tokens` -> `userAddedTokensAtom:v1`
2. User-added lists: `redux_localstorage_simple_lists` -> `allTokenListsInfoAtom:v2`
3. Also, favorite tokens were moved from `redux_localstorage_simple_user.favouriteTokens` to `favouriteTokensAtom`. I didn't migrate them because it looks like it doesn't worth it.
4. `redux_localstorage_simple_user.pairs` was just removed, now `tokensStateAtom` performs its functions. 

  # To Test

1. Open https://dev.swap.cow.fi/
2. Add some custom tokens and lists
3. Keep some newly added lists enabled and some disabled
4. Run `downloadLocalStorage()` in the console (see the snipped below). It will download a file with the localStorage state
5. Open Dapp from this PR
6. Clean all browser data for this URL: Console -> Application -> Storage -> "Clear site data"
7. Copy the content of the file from step 4 and run in the console: `var state = <PUT_THE_FILE_CONTENT>`, press Enter
8. Run the code in the console: `localStorage.clear(); var stateObj = JSON.parse(state); Object.keys(stateObj).forEach(key => localStorage.setItem(key, stateObj[key]))`. It will apply the localStorage state from dev.swap.cow.fi into the current tab
9. Reload the page
- [ ] TOKENS_REFACTORING_MIGRATED should be added in the localStorage
- [ ] User-added tokens should be the same as in dev.swap.cow.fi
- [ ] User-added lists should be the same as in dev.swap.cow.fi including enabled/disabled state
- [ ] Favourite tokens should be reset to defaults
- [ ] `redux_localstorage_simple_lists` should be removed from localStorage
- [ ] `redux_localstorage_simple_user.favouriteTokens` should be removed from localStorage
- [ ] `redux_localstorage_simple_user.tokens` should be removed from localStorage
- [ ] `redux_localstorage_simple_user.pairs` should be removed from localStorage
10. Open the PR in incognito mode
- [ ] there should not be user-added tokens
- [ ] there should not be user-added lists

### Download localStorage snippet:
```
function downloadLocalStorage() {
    const storageObj = JSON.stringify(Object.keys(localStorage).reduce((acc, key) => {
        acc[key] = localStorage.getItem(key)

        return acc
    }, {}));
    
    const a = document.createElement('a');
    a.style.display = 'none';
    a.href = "data:text/json;charset=utf-8," + encodeURIComponent(JSON.stringify(storageObj))
    // the filename you want
    a.download = 'todo-1.json';
    document.body.appendChild(a);
    a.click();
}
```

### CODE REVIEW

File: libs/tokens/src/pure/TokenLogo/index.tsx [hunk 1]
```diff
@@ -17,6 +17,7 @@ const TokenLogoWrapper = styled.div`
   display: inline-block;
   background: var(--cow-container-bg-01);
   border-radius: 50%;
+  overflow: hidden;
```
Comment (shoom3301) [inline libs/tokens/src/pure/TokenLogo/index.tsx:20]: Just noticed we have some square tokens logos

**Review by alfetopito — COMMENTED** [review]
Very nice, loved the migration!

One of the tests failed, though.

The legacy `simple_lists` key is still there.
image

**Comment by shoom3301** [comment]
@alfetopito it looks like an outdated state in the dev tools. Could you please try to click the refresh button?

image

**Review by anxolin — COMMENTED** [review]
I'm not sure if I follow correctly the steps, but it didn't work for me


1. I added SHIT token in DEV: https://etherscan.io/token/0x4e4a47cAC6A28A62dcC20990ed2cdA9BC659469F

image


2. Exported all my local history in DEV (after adding the SHIT token)

3. Cleared the local storage in this PR

4. Imported the local storage

4. Try to navigate to SHIT. I get an IMPORT MODAL for some other Token I don't know about: SUZUME


image

**Comment by shoom3301** [comment]
@anxolin thank you! Nice catch!
It's because I didn't add a strict search for tokens auto-importing.
The fix: https://github.com/cowprotocol/cowswap/pull/3267/commits/31b33ae2e19bf45767cf548da71815964c4e858d

**Comment by alfetopito** [comment]
> @alfetopito it looks like an outdated state in the dev tools. Could you please try to click the refresh button?
> image

I tried, didn't help:

https://github.com/cowprotocol/cowswap/assets/43217/aefa850d-25d6-495f-b8ab-88ced4c25123

**Comment by shoom3301** [comment]
@alfetopito thanks!
The problem was in `8. Run the code in the console` script. Updated it, works well now

File: libs/tokens/src/hooks/tokens/useSearchNonExistentToken.ts [hunk 2]
```diff
@@ -23,7 +23,9 @@ export function useSearchNonExistentToken(tokenId: string | null): TokenWithLogo
     return (
       [foundToken.inactiveListsResult, foundToken.externalApiResult, foundToken.blockchainResult]
         .flat()
-        .filter(isTruthy)[0] || null
+        .filter((token) => {
+          return !!token && doesTokenMatchSymbolOrAddress(token, inputTokenToSearch)
+        })[0] || null
```
Comment (alfetopito) [inline libs/tokens/src/hooks/tokens/useSearchNonExistentToken.ts:28]: Tiny nitpick: remove `{ return }`

**Review by alfetopito — APPROVED** [review]

**Pull request merged**